### PR TITLE
Feat/own profile screen

### PR DESCRIPTION
**Pull request by margdsq**
This pull request includes several changes to the profile screen, home screen, and navigation setup. The most important changes include adding tests for the profile screen, modifying the home screen to include navigation to the profile screen, and updating the navigation to include the profile screen.

### Profile Screen Enhancements and Tests:

* [`__tests__/UI-tests/Profile_Screen_test.tsx`](diffhunk://#diff-283a922cb9af7c81b27f6e8411215e66383dfb6d26740d5daeaf91786a73e9d1R1-R75): Added tests to verify the profile screen renders correctly, handles image picker functionality, and logs out the user.
* [`app/screens/home/profile_screen.tsx`](diffhunk://#diff-70a284082f8cb98eea751e4e159aa2a50b8a472883b55b9569e0e4bb175ef410R1-R201): Implemented the profile screen with functionalities for changing email, changing password, logging out, and picking an image.

### Home Screen Modifications:

* [`app/screens/home/home_screen.tsx`](diffhunk://#diff-6526ae8a24a7d12b3c578f5add3c25b6adff57eec7ca6c7c8010da92c01e1b59R21): Added console log for `uid` in `useEffect` and navigation to the profile screen on a button click. [[1]](diffhunk://#diff-6526ae8a24a7d12b3c578f5add3c25b6adff57eec7ca6c7c8010da92c01e1b59R21) [[2]](diffhunk://#diff-6526ae8a24a7d12b3c578f5add3c25b6adff57eec7ca6c7c8010da92c01e1b59R43)

### Navigation Updates:

* [`navigation/Navigation.tsx`](diffhunk://#diff-f2ebc3106c47c607114011cb93fd410dc27e8a5e8d9b82db5cfa8da784151462R17): Imported the profile screen and added it to the navigation stack. [[1]](diffhunk://#diff-f2ebc3106c47c607114011cb93fd410dc27e8a5e8d9b82db5cfa8da784151462R17) [[2]](diffhunk://#diff-f2ebc3106c47c607114011cb93fd410dc27e8a5e8d9b82db5cfa8da784151462R99-R108)

### Miscellaneous Changes:

* [`app/screens/welcome/final_screen.tsx`](diffhunk://#diff-91c9197d7457990c2cb5397c304500ec7e038ff5da9cd1e3c96bea9a4899a2a3R54): Added sign-out functionality when continuing as a guest.
* [`app/screens/welcome/final_screen.tsx`](diffhunk://#diff-91c9197d7457990c2cb5397c304500ec7e038ff5da9cd1e3c96bea9a4899a2a3R4): Imported `auth` from `firebase/Firebase`.

### What needs to be done : 

I still have to implement with Firebase the change of the profile picture, but I need Roman's PR to do so, hence it will be implemented real soon.

### CODE REVIEW

File: __tests__/UI-tests/Profile_Screen_test.tsx
Comment (Tristan-C789) [inline __tests__/UI-tests/Profile_Screen_test.tsx:1]: The name of the file should be `ProfileScreen-test.tsx` to follow the same standard as others.

File: app/screens/home/home_screen.tsx
Comment (Tristan-C789) [inline app/screens/home/home_screen.tsx:1]: Are the logs still needed ? They seem a lot like debugging logs.

File: app/screens/home/profile_screen.tsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+import { Dimensions } from "react-native";
+import { ThemedTextButton } from "@/components/theme/ThemedTextButton";
+import { BottomBar } from "@/components/navigation/BottomBar";
+import { Alert } from "react-native";
+import { Icon } from "react-native-elements";
+import { auth } from "@/firebase/Firebase"; 
+import { updateEmail, sendPasswordResetEmail} from "firebase/auth";
+
+
+//TODO : change the colors for light mode
```
Comment (Tristan-C789) [inline app/screens/home/profile_screen.tsx:19]: What's this comment corresponding to ?

File: app/screens/home/profile_screen.tsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+const { width, height } = Dimensions.get("window");
+
+export default function ProfileScreen({user, navigation, firestoreCtrl}: any) {
+
+  const currentUser = auth.currentUser;
+  const username = currentUser?.displayName;
+  const email = currentUser?.email;
+  //const username = "test";
+  const [image, setImage] = React.useState<string | null>(null);
+  const [isLoggedIn, setIsLoggedIn] = React.useState<Boolean>(auth.currentUser ? true : false);
```
Comment (Tristan-C789) [inline app/screens/home/profile_screen.tsx:29]: Perhaps this should be `...(auth.currentUser == null);` for better clarity ?

File: app/screens/welcome/final_screen.tsx [hunk 3]
```diff
@@ -50,6 +51,7 @@ export default function WelcomeFinalScreen({ navigation, firestoreCtrl }: any) {
               index: 0,
               routes: [{ name: "Home" }],
             });
+            auth.signOut();
```
Comment (Tristan-C789) [inline app/screens/welcome/final_screen.tsx:54]: Is that really necessary ?

**Review by Tristan-C789 — CHANGES_REQUESTED** [review]
Great PR overall! I've requested some minor changes on some files, but those should be resolved quickly.

File: app/screens/welcome/final_screen.tsx (hunk 3 above)
Comment (margdsq) [reply app/screens/welcome/final_screen.tsx:54]: Yes actually. If you had recently logged in and you go as guest, you were still logged in which doesn't make any sense

File: app/screens/home/profile_screen.tsx (hunk 1 above)
Comment (margdsq) [reply app/screens/home/profile_screen.tsx:19]: Just because I didn't finished the UI for light screen

**Review by Tristan-C789 — APPROVED** [review]
You successfully addressed the issues I pointed out, thanks!

**Pull request merged**